Loan.ts

```diff
@@ -0,0 +1,32 @@
+import FineractAPI from './FineractAPI';
+
+export interface LoanConfig{
+    offset?:Number,
+    limit?:Number,
+    orderBy?:String,
+    sortBy?:String,
+    officeId?:Number,
+    underHierarchy?:String,
+    accountNo?:String,
+    externalId?:String,
+    sqlSearch?:String
+}
+export default class Loan{
+
+fineract_obj:FineractAPI
+constructor(fineract_obj:FineractAPI)
+{
+   this.fineract_obj=fineract_obj;
+}
+
+async list_loan(config:LoanConfig):Promise<any> {
```

Comment:
The return type of this function should be an array of LoanObjectInterface

```suggestion
async list_loan(config:LoanConfig):Promise<[LoanObjectInterface]> 
```
and LoanObjectInterface should represent the following data structure

```json
 {
      "id": 1,
      "accountNo": "000000001",
      "status": {
        "id": 300,
        "code": "loanStatusType.active",
        "value": "Active",
        "pendingApproval": false,
        "waitingForDisbursal": false,
        "active": true,
        "closedObligationsMet": false,
        "closedWrittenOff": false,
        "closedRescheduled": false,
        "closed": false,
        "overpaid": false
      },
      "clientId": 1,
      "clientName": "Change To IndividualName",
      "clientOfficeId": 1,
      "loanProductId": 1,
      "loanProductName": "AgriCredit",
      "loanProductDescription": "test",
      "loanType": {
        "id": 1,
        "code": "loanType.individual",
        "value": "Individual"
      },
      "currency": {
        "code": "USD",
        "name": "US Dollar",
        "decimalPlaces": 2,
        "displaySymbol": "$",
        "nameCode": "currency.USD",
        "displayLabel": "US Dollar ($)"
      },
      "principal": 10000,
      "termFrequency": 12,
      "termPeriodFrequencyType": {
        "id": 2,
        "code": "termFrequency.periodFrequencyType.months",
        "value": "Months"
      },
      "numberOfRepayments": 12,
      "repaymentEvery": 1,
      "repaymentFrequencyType": {
        "id": 2,
        "code": "repaymentFrequency.periodFrequencyType.months",
        "value": "Months"
      },
      "interestRatePerPeriod": 2,
      "interestRateFrequencyType": {
        "id": 2,
        "code": "interestRateFrequency.periodFrequencyType.months",
        "value": "Per month"
      },
      "annualInterestRate": 24,
      "amortizationType": {
        "id": 1,
        "code": "amortizationType.equal.installments",
        "value": "Equal installments"
      },
      "interestType": {
        "id": 0,
        "code": "interestType.declining.balance",
        "value": "Declining Balance"
      },
      "interestCalculationPeriodType": {
        "id": 1,
        "code": "interestCalculationPeriodType.same.as.repayment.period",
        "value": "Same as repayment period"
      },
      "transactionProcessingStrategyId": 2,
      "timeline": {
        "submittedOnDate": [
          2012,
          6,
          1
        ],
        "submittedByUsername": "kred",
        "submittedByFirstname": "App",
        "submittedByLastname": "Administrator",
        "approvedOnDate": [
          2012,
          6,
          1
        ],
        "approvedByUsername": "kred",
        "approvedByFirstname": "App",
        "approvedByLastname": "Administrator",
        "expectedDisbursementDate": [
          2012,
          6,
          1
        ],
        "actualDisbursementDate": [
          2012,
          6,
          1
        ],
        "disbursedByUsername": "kred",
        "disbursedByFirstname": "App",
        "disbursedByLastname": "Administrator",
        "expectedMaturityDate": [
          2013,
          6,
          1
        ]
      },
      "linkedAccount":{
    	"id":1,
    	"accountNo":"000000001"
      },
      "summary": {
        "currency": {
          "code": "USD",
          "name": "US Dollar",
          "decimalPlaces": 2,
          "displaySymbol": "$",
          "nameCode": "currency.USD",
          "displayLabel": "US Dollar ($)"
        },
        "principalDisbursed": 10000,
        "principalPaid": 3073.07,
        "principalWrittenOff": 0,
        "principalOutstanding": 6926.93,
        "principalOverdue": 5999.92,
        "interestCharged": 1347.15,
        "interestPaid": 709.33,
        "interestWaived": 0,
        "interestWrittenOff": 0,
        "interestOutstanding": 637.82,
        "interestOverdue": 619.28,
        "feeChargesCharged": 0,
        "feeChargesDueAtDisbursementCharged": 0,
        "feeChargesPaid": 0,
        "feeChargesWaived": 0,
        "feeChargesWrittenOff": 0,
        "feeChargesOutstanding": 0,
        "feeChargesOverdue": 0,
        "penaltyChargesCharged": 0,
        "penaltyChargesPaid": 0,
        "penaltyChargesWaived": 0,
        "penaltyChargesWrittenOff": 0,
        "penaltyChargesOutstanding": 0,
        "penaltyChargesOverdue": 0,
        "totalExpectedRepayment": 11347.15,
        "totalRepayment": 3782.4,
        "totalExpectedCostOfLoan": 1347.15,
        "totalCostOfLoan": 709.33,
        "totalWaived": 0,
        "totalWrittenOff": 0,
        "totalOutstanding": 7564.75,
        "totalOverdue": 6619.2,
        "overdueSinceDate": [
          2012,
          11,
          1
        ]
      },
      "feeChargesAtDisbursementCharged": 0,
      "inArrears":false,
      "isNPA":false
    }
```
as mentioned in the example response of https://kred.gitlab.io/cbs-api-doc/#loans_list